islandora_pathauto.module

```diff
@@ -141,45 +141,108 @@ function islandora_pathauto_islandora_object_purged($pid) {
 }
 
 /**
- * Implements hook_pathauto_bulkupdate().
+ * Gets the islandora_pathauto module results. 
+ *
+ * This is based on the query language specified by
+ * islandora_pathauto_query_language.
+ *
+ *
+ * @return array
+ *   results of object, title, cmodel
  */
-function islandora_pathauto_pathauto_bulkupdate() {
-  module_load_include('inc', 'islandora', 'includes/utilities');
-  $query = 'SELECT $object $title $cmodel
-        FROM <#ri>
-        WHERE {
-            $object <fedora-model:label> $title ;
-            <fedora-model:hasModel> $cmodel ;
-            <fedora-model:state> <fedora-model:Active> .
-            FILTER(!sameTerm($cmodel, <info:fedora/fedora-system:FedoraObject-3.0>))
-            }
-            ORDER BY $title';
-  $tuque = islandora_get_tuque_connection();
-  if ($tuque) {
+function islandora_pathauto_get_module_results() {
+  // Step 1, get the enabled models.
+  $models = variable_get('islandora_pathauto_selected_cmodels', array());
+  // Depending on the query language, get the result records that match the
+  // enabled models.
+  $query_language = variable_get('islandora_pathauto_query_language', 'islandora_basic_collection_sparql_query_backend');
+  if ($query_language == 'islandora_basic_collection_sparql_query_backend') {
+    $first_contentmodel = array_shift($models);
+    $unions = '';
+    foreach ($models as $contentmodel) {
+      $unions .= 'UNION {
+              $object <fedora-model:label> $title ;
+              <fedora-model:hasModel> $cmodel ;
+              <fedora-model:state> <fedora-model:Active> .
+              FILTER(sameTerm($cmodel, <info:fedora/' . $contentmodel . '>))}
+              }';
+    }
+    $query = 'SELECT $object $title $cmodel
+          FROM <#ri>
+          WHERE { {
+              $object <fedora-model:label> $title ;
+              <fedora-model:hasModel> $cmodel ;
+              <fedora-model:state> <fedora-model:Active> .
+              FILTER(sameTerm($cmodel, <info:fedora/' . $first_contentmodel . '>))
+              }' . $unions . '
+              ORDER BY $title';
+    $tuque = islandora_get_tuque_connection();
+    if ($tuque) {
+      try {
+        $results = $tuque->repository->ri->query($query, 'sparql');
+        return $results;
+      }
+      catch (Exception $e) {
+        return array();
+      }
+    }
+    else {
+      IslandoraTuque::getError();
+    }
+  }
+  elseif ((module_exists('islandora_solr') == TRUE) && ($query_language == 'islandora_solr_query_backend')) {
+    module_load_include('inc', 'islandora_solr_search', 'includes/utilities');
+    $query_processor = new IslandoraSolrQueryProcessor();
+    $query_processor->solrQuery = 'RELS_EXT_hasModel_uri_s:(*' . str_replace(":", "\:", implode(" OR *", $models)) . ')';
+    $query_processor->solrStart = 0;
```

Comment:
You'll need a way to page the results similar to here: https://github.com/Islandora/islandora_pathauto/blob/7.x/islandora_pathauto.module#L251-L256

...otherwise you'll only be update the first <islandora_pathauto_query_limit>